src/Compilers/Test/Core/Diagnostics/DiagnosticDescription.cs

```diff
@@ -463,20 +463,25 @@ public override string ToString()
 
         private static ImmutableArray<string> GetFormatSpecifiers(string messageFormat)
         {
-            ImmutableArray<string> specifiers = default;
+            var specifiers = ImmutableArray<string>.Empty;
             if (Regex.Matches(messageFormat, @"\{\d+(:\d+)?\}") is { Count: > 0 } matches)
             {
                 var builder = ArrayBuilder<string>.GetInstance();
                 foreach (Match match in matches)
                 {
+                    // We use 0 as the position specifier, regardless of what it was in the original format string,
+                    // because we format diagnostic arguments one at a time so we cannot have a position specifier greater than 0
+                    var posSpecifier = "0";
+                    string fmtSpecifier = "";
                     if (match.Groups.Count > 1 && match.Groups[1].Success)
                     {
-                        builder.Add($@"{{0{match.Groups[1].Value}}}");
-                    }
-                    else
-                    {
-                        builder.Add("{0}");
+                        fmtSpecifier = match.Groups[1].Value;
                     }
+
+                    builder.Add(
+                        string.IsNullOrEmpty(fmtSpecifier) ?
```

Comment:
Is this check actually necessary? If it's `null` or `""`, then it should just format nothing.